src/main/java/uk/gov/hmcts/reform/cpo/service/impl/CasePaymentOrdersServiceImpl.java

```diff
@@ -64,4 +66,16 @@ private void validateCasePaymentOrderQueryFilter(final CasePaymentOrderQueryFilt
                 "case-payment-orders cannot filter case payments orders by both id and cases-id.");
         }
     }
+
+    @Override
+    public void deleteCasePaymentOrdersByIds(List<UUID> ids) throws CasePaymentIdentifierException {
+        casePaymentOrdersRepository.deleteByUuids(ids);
+        casePaymentOrdersRepository.deleteAuditEntriesByUuids(ids);
```

Comment:
is this transactional? I don't see @transactional on the corresponding controller or service method

Reply:
It is on the `CasePaymentOrdersRepository` interface

Reply:
yep saw that, but what I meant is because the transactionality is on the repository, let's say there's an error in the deletion of the audits entries, will the deletion of the cpo be rolled back as well?

Reply:
Ok, I understand now -  looks like I need to write some tests for this scenario, and refactor accordingly

Reply:
fixed in 791effd